File: 4th Semester/Design and Programming Environments/labs/5th lab/C#/BasketModel/Purchase.cs
using System;

namespace BasketModel
{
    [Serializable]
    public class Purchase : Entity<long>
    {
        private String client;
        private long game;
        private int seats;
        private String address;
        public Purchase(String client, long game, int seats, String address) {
            this.client = client;
            this.game = game;
            this.seats = seats;
            this.address = address;
        }

        public String getClient() {
            return client;
        }

        public void setClient(String client) {
            this.client = client;
        }

        public long getGame() {
            return game;
        }

        public void setGame(long game) {
            this.game = game;
        }

        public int getSeats() {
            return seats;
        }

        public void setSeats(int seats) {
            this.seats = seats;
        }

        public String getAddress() {
            return address;
        }

        public void setAddress(String address) {
            this.address = address;
        }
    }
}

File: 4th Semester/Design and Programming Environments/labs/8th & 9th lab/Java/basket_orm/BasketModel/src/main/java/bonti/model/Purchase.java
package bonti.model;

import jakarta.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "Purchases")
public class Purchase implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String client;

    private Long game;

    private int seats;

    private String address;

    public Purchase() {}

    public Purchase(Long id, String client, Long game, int seats, String address) {
        this.id = id;
        this.client = client;
        this.game = game;
        this.seats = seats;
        this.address = address;
    }

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getClient() { return client; }
    public void setClient(String client) { this.client = client; }

    public Long getGame() { return game; }
    public void setGame(Long game) { this.game = game; }

    public int getSeats() { return seats; }
    public void setSeats(int seats) { this.seats = seats; }

    public String getAddress() { return address; }
    public void setAddress(String address) { this.address = address; }
}
